### PR TITLE
Moves `cli` to TypeScript and uses Inquirer.js

### LINKED ISSUE #76: Convert Project into TypeScript
The project was started using pure JavaScript. We're migrating the code to TypeScript, to make it more robust and easier to maintain.

Any new files added **must** use TypeScript.

- [x] `index.js`
- [x] `lib/Config.js`
- [x] `lib/GradleWrapper.js`
- [x] `lib/Log.js`
- [x] `lib/TwaGenerator.js`
- [x] `lib/TwaManifest.js`
- [x] `lib/util.js`
- [x] `lib/androidSdk/AndroidSdkTools.js`
- [x] `lib/androidSdk/AndroidSdkInstaller.js`
- [x] `lib/jdk/JdkHelper.js`
- [x] `lib/jdk/JdkInstaller.js`
- [x] `lib/jdk/KeyTool.js`
- [ ] `cli/index.js`
- [ ] `cli/cmds/init.js`
- [ ] `cli/cmds/build.js`
- [ ] `cli/cmds/update.js`
- [ ] `cli/cmds/help.js`
- [x] `spec/lib/ConfigSpec.js`
- [x] `spec/lib/TwaManifestSpec.js`
- [x] `spec/lib/utilSpec.js` 
- [x] `spec/lib/androidSdk/AndroidSdkToolsSpec.js`
- [x] `spec/lib/jdk/JdkHelperSpec.js`

### LINKED ISSUE #79: Replace `prompt` with `inquirer.js`
[`prompt`](https://github.com/flatiron/prompt) seems to be largely unmaintained, with the last comming being around 3 years ago and many open issues. TypeScript is also not supported (which will affect #76)

[`inquirer.js`](https://www.npmjs.com/package/inquirer) seems better maintained and has [TypesScript types](https://www.npmjs.com/package/@types/inquirer).

### PR DESCRIPTION
**Pull request by andreban**
- Moves the remaining JS files in cli/ to TypeScript
- Replaces 'prompt' with 'Inquirer.js'

Closes #76 
Closes #79

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -14,9 +14,7 @@
  *  limitations under the License.
  */
 
-'use strict';
```
Comment (PEConn) [inline src/index.ts]: how come we're removing this?

File: src/lib/Config.ts [hunk 2]
```diff
@@ -18,11 +18,10 @@ import {promisify} from 'util';
 import * as fs from 'fs';
 import * as path from 'path';
 import * as os from 'os';
-import userPrompt = require('prompt');
+import * as inquirer from 'inquirer';
```
Comment (PEConn) [inline src/lib/Config.ts]: Not essential, but I'd still prefer we be precise with our imports.

File: src/lib/TwaGenerator.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ import {template} from 'lodash';
 import {promisify} from 'util';
 import {TwaManifest} from './TwaManifest';
 import Jimp = require('jimp');
+import Log from './Log';
```
Comment (PEConn) [inline src/lib/TwaGenerator.ts]: TypeScript question - why do some of these have `{}` and some don't?

File: src/cli/inputHelpers.ts [hunk 4]
```diff
@@ -14,8 +14,26 @@
...
 
-declare module 'prompt' {
-  import prompt = require('prompt');
-  // TODO(andreban): Remove usage of any
-  export function get(schema: any): any; // eslint-disable-line @typescript-eslint/no-explicit-any
+import Color = require('color');
+import {isWebUri} from 'valid-url';
+
+export function validatePassword(input: string): boolean {
+  return input.length > 0;
```
Comment (PEConn) [inline src/cli/inputHelpers.ts]: Sure we don't want something longer?

File: src/cli/inputHelpers.ts [hunk 5]
```diff
@@ -14,8 +14,26 @@
...
+}
+
+export function notEmpty(input: string): boolean {
+  return input.length > 0;
```
Comment (PEConn) [inline src/cli/inputHelpers.ts]: `input.trim().length > 0` ?

File: src/cli/inputHelpers.ts [hunk 6]
```diff
@@ -14,8 +14,26 @@
  *  limitations under the License.
  */
 
-declare module 'prompt' {
-  import prompt = require('prompt');
-  // TODO(andreban): Remove usage of any
-  export function get(schema: any): any; // eslint-disable-line @typescript-eslint/no-explicit-any
+import Color = require('color');
```
Comment (PEConn) [inline src/cli/inputHelpers.ts]: And again with the imports questions, why is this `require`?

File: src/cli/cmds/help.ts [hunk 7]
```diff
@@ -14,9 +14,6 @@
  *  limitations under the License.
  */
 
-'use strict';
-
-async function help() {
+export async function help(): Promise<void> {
+  return;
```
Comment (PEConn) [inline src/cli/cmds/help.ts]: help is a no-op?

File: src/cli/cmds/init.ts [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+}
+
+async function confirmTwaConfig(twaManifest: TwaManifest): Promise<TwaManifest> {
+  const result = await inquirer.prompt([
+    {
+      name: 'host',
+      type: 'input',
+      message: 'Domain being opened in the TWA:',
+      default: twaManifest.host,
+      validate: notEmpty,
```
Comment (PEConn) [inline src/cli/cmds/init.ts]: We could check this is a valid URL (future CL though).

File: src/cli/cmds/init.ts [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+    }, {
+      name: 'name',
+      type: 'input',
+      message: 'Name of the application:',
+      default: twaManifest.name,
+      validate: notEmpty,
```
Comment (PEConn) [inline src/cli/cmds/init.ts]: Similarly, we could check that this and `launcherName` aren't too long.

File: src/cli/cmds/init.ts [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+    }, {
+      name: 'launcherName',
+      type: 'input',
+      message: 'Name to be shown on the Android Launcher:',
+      default: twaManifest.launcherName,
+      validate: notEmpty,
+    }, {
+      name: 'themeColor',
+      type: 'input',
+      message: 'Color to be used for the status bar:',
```
Comment (PEConn) [inline src/cli/cmds/init.ts]: It's maybe worth putting an example of the value format (is it #232323, just 232323, either, both?).

File: src/index.ts (hunk 1 above)
Comment (andreban) [reply src/index.ts]: We have `strict: true` on `tsconfig.json`, so TypeScript will add `'use string' "to the output JS for us.

File: src/lib/TwaGenerator.ts (hunk 3 above)
Comment (andreban) [reply src/lib/TwaGenerator.ts]: On the Log, we do `export default class Log`. on TwaManifest, only `export class TwaManifest`.

File: src/cli/inputHelpers.ts (hunk 4 above)
Comment (andreban) [reply src/cli/inputHelpers.ts]: I need to look at the requirements for keytool and replicate there. Doing on a following CL.

File: src/cli/cmds/help.ts (hunk 7 above)
Comment (andreban) [reply src/cli/cmds/help.ts]: For now... Need to add actual help messages (I'll do in another PR. Focusing this one on translating JS => TS)

File: src/cli/inputHelpers.ts (hunk 6 above)
Comment (andreban) [reply src/cli/inputHelpers.ts]: I don't fully understand the require one, but it's related to the underlying library being written in JavaScript. Using `import * as Color from 'color'`, `import {Color} from 'color'` or `import Color from 'color'` reports the error below:
```
Module '"/Users/andreban/Projects/twa-bootstrap/node_modules/@types/color/index"' can only be default-imported using the 'esModuleInterop' flagts(1259)

index.d.ts(135, 1): This module is declared with using 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.
```

File: src/cli/cmds/init.ts (hunk 8 above)
Comment (andreban) [reply src/cli/cmds/init.ts]: This one is not a full URL, host a host (I agree we should check)

Comment (andreban) [reply src/cli/cmds/init.ts]: #31 is the one I created for those.

File: src/cli/cmds/init.ts (hunk 9 above)
Comment (andreban) [reply src/cli/cmds/init.ts]: Another good one for #31 (Adding here so I get the mentions in the issue)

File: src/cli/cmds/init.ts (hunk 10 above)
Comment (andreban) [reply src/cli/cmds/init.ts]: The default value shows the correct format (`#232323`). Is this enough?

Comment (andreban) [reply src/cli/cmds/init.ts]: Also, we accept a wide range of formats. Including `blue`, `rgb(0, 0, 255)`, etc.

Comment (PEConn) [reply src/cli/cmds/init.ts]: Yeah, that's fine.

File: src/lib/Config.ts (hunk 2 above)
Comment (andreban) [reply src/lib/Config.ts]: I'm a bit divided in the case where a library has many related methods. 

When we import with `import * as inquirer from 'inquirer'`, then we get to use `inquirer.prompt(...)` in the code, which makes it clear where prompt is coming from.

When we import with `import {prompt} from 'inquirer'`, the call in the code becomes just `prompt`, which is less obvious where it comes from, unless you scan the imports. 

WDYT?

File: src/cli/inputHelpers.ts (hunk 5 above)
Comment (andreban) [reply src/cli/inputHelpers.ts]: +1. done

**Pull request merged**